pkg/machines/app.jsx

```diff
@@ -114,10 +115,14 @@ class App extends React.Component {
         const createVmAction = <CreateVmAction {...properties} mode='create' />;
         const importDiskAction = <CreateVmAction {...properties} mode='import' />;
         const vmActions = <div> {createVmAction} {importDiskAction} </div>;
+        const resources = [...storagePools, ...networks, ...nodeDevices, ...interfaces, ...vms];
```

Comment:
Your solution works but I was thinking about simplifying the code like this: 
Let me know if you see any issues with my implementation idea.

So you initiallize all resource initial state to undefined.
https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/reducers.js#L102
So replace state || [] -> state || undefined

Then if you need to also make sure to create the array when adding the first resource.
You can then remove completely https://github.com/cockpit-project/cockpit/pull/12913/commits/1c0419957355e52bcc934ec7ba818f1eb683e969 since we are not interested in each resource seperately.
And you need to replace this condition here, checkign that none of the storagePools, networks, ... are undefined.
What do you think?

Reply:
> You can then remove completely [1c04199](https://github.com/cockpit-project/cockpit/commit/1c0419957355e52bcc934ec7ba818f1eb683e969) since we are not interested in each resource separately.

I actually am interested in each resource separately. The problem would be that some resource like "nodeDevices" would be considered not loaded as long as it's "undefined", but as soon as I insert first node device into array, the resources would be considered loaded, although it takes some time to load every single resource.
For example there is usually hundreds of node devices, so as soon as I insert the first device into array, we would consider it loaded and show page content to the user, but the loading of other hundreds of node devices would be still ongoing.

Alternative solution would be to to initialize "nodeDevices" to undefined as you propose, then poll all of the nodedevices separately into some array, and then once every single nodeDevice is loaded, replace the undefined with this array with single go.

Reply:
Oh right, I now got what you are actually doing. Thanks!